### DISCUSSION TITLE
About installing 'CLOC'. Help needed

### DISCUSSION THREAD

**Discussion by Taffy91** [Cyfrin/security-and-auditing-full-course-s23#38] [Q&A] [answered]
I have tried to use the package manager to install CLOC as required in Section 3 of the auditing course.  I am a Windows user on a Linux subsystem using Ubuntu. Upon typing in 'sudo apt install cloc' as the package manager instructed, I am then asked for a password in the terminal. The issue is, i cannot type anything in the terminal after pressing enter after typing 'sudo apt install cloc' in my terminal. What am I to do if this happens everytime after killing the terminal an restarting this process? I just want to install CLOC on my system. Thanks in advance

**Comment by PatrickAlphaC** [comment]
> i cannot type anything in the terminal after pressing enter after typing

Usually, it'll type your password, just not show you it on screen. Can you just type your password and press enter? You password will be entered, just not shown.

**Reply by cromewar** [reply]
@Taffy91  That happens on most linux distributions, the super user password is not shown when typing, you would probably be fine shit typing your password as @PatrickAlphaC said.

Also you can cancel any running command with `ctrl + c` or `ctrl + z`. Give it a try.

**Reply by Taffy91** [reply]
🤣 thanks you both. It really as simple as that, good thing I asked. I spent hours on that one thing, I appreciate this tons thanks again.

**Reply by Taffy91** [reply]
@cromewar hi again. Is the sudo password my login password for my laptop account? I tried typing in my laptop password and the terminal said, 'Sorry, try again.' I only typed in my laptop password because upon research different sources said that the sudo password is my normal laptop login password. Also,  is there a way to reset my sudo password? I would like to do possibly reset my password because I'm honestly just eager to get on with the course. Different things have been set backs to me completing these courses, but I really am happy that you all did a smart contract auditing course because that's where my passion lies. Thanks in advance.

**Reply by PatrickAlphaC** [reply]
It should be your laptop password. Are you using an AI helper? Usually something like phind or chatgpt can help debug these.

**Reply by cromewar** [reply]
@Taffy91, yet your laptop password must be the one. Unless you have multiple users and the one you have is not and admin or your computer is from work and you have sys admin.

**Reply by Taffy91** [reply, marked as answer]
thank you, I neglected to mention that I solved this issue by resetting the password. Thanks a lot for your response. I do appreciate it. And sorry for the late reply, my internet among other things was down.
